pkgs/tools/admin/tigervnc/default.nix

```diff
@@ -31,14 +45,16 @@ stdenv.mkDerivation rec {
     })
   ];
 
-  postPatch = ''
+  postPatch = lib.optionalString stdenv.isLinux ''
     sed -i -e '/^\$cmd \.= " -pn";/a$cmd .= " -xkbdir ${xkeyboard_config}/etc/X11/xkb";' unix/vncserver/vncserver.in
     fontPath=
     substituteInPlace vncviewer/vncviewer.cxx \
        --replace '"/usr/bin/ssh' '"${openssh}/bin/ssh'
 
     cp unix/xserver21.1.1.patch unix/xserver211.patch
     source_top="$(pwd)"
+  '' + lib.optionalString stdenv.isDarwin ''
+    substituteInPlace release/makemacapp.in --replace 'hdiutil' '/usr/bin/hdiutil'
```

Comment:
Why do we want this?

Reply:
For darwin, we need the generated .app, which gets built when `make dmg` is run, which itself calls `hdiutil`, which isn't (to my knowledge) available within nix. While elsewhere in the PR I use `undmg` to extract that `.dmg`'s `.app`, I'm not sure how else to build it.

A slightly more intrusive approach would be to actually edit https://github.com/TigerVNC/tigervnc/blob/master/release/makemacapp.in to remove the .dmg generation lines altogether and just copy the `.app` directly, but I assumed that'd be more likely to break on a version bump.

Reply:
We should change the build script to not generate the dmg archive and then unpack it again. Thats a waste of compute.